src/spec-node/singleContainer.ts

```diff
@@ -291,6 +291,17 @@ export async function findDevContainer(params: DockerCLIParameters | DockerResol
 	return details.filter(container => container.State.Status !== 'removing')[0];
 }
 
+export async function extraRunArgs(params: DockerCLIParameters | DockerResolverParameters, config: DevContainerFromDockerfileConfig | DevContainerFromImageConfig) {
```

Comment:
Can the same be done for the Docker Compose case? (dockerCompose.ts)

Reply:
@chrmarti that's a really good spot, I hadn't seen there was a separate code path there, I'll take a look.

Reply:
@chrmarti I've had a look through the code and the documentation for `docker compose`, and I'm nor sure what the most sensible approach would be here.

The 'normal' `docker-compose.yml` file can specify that a specific service needs a gpu capability, and will fail to start if one isn't available. This is the same situation as when using `Dockerfile`, and specifying the extra arguments, and so not a great experience for developers on mixed hardware.

The only way I can see to fix this is to modify the `docker-compose.yml` before it is used - either adding or removing the gpu capability. Removing is seems slightly more logical, as then it's clear which service should have gpu access if it's available.

So my current thoughts are that the logic when using docker-compose should be:
1. If the devontainer.json lists gpu's as required,
2. and the docker-compose.yml has services requiring gpu capabilities,
3. and the nvidia extensions are not available,
4. then create a temporary copy of the yaml without the gpu capabilities, and launch that instead.

How does that sound?

Reply:
I've implemented the flow described above - let me know hoe it looks

Reply:
We use overlay docker-compose.yml files in other cases. So you could probably (I haven't tried) enable gpu support in `generateFeaturesComposeOverrideContent()` if required instead of removing it if it is already there. Being additive this would be closer to the single container case.